src/main/java/ivanov/springbootintro/service/impl/BookServiceImpl.java

```diff
@@ -44,42 +59,45 @@ public List<BookDto> findAll(Pageable pageable) {
 
     @Override
     public BookDto findById(Long id) {
-        Book book = bookRepository.findById(id).orElseThrow(
-                () -> new EntityNotFoundException("Can't find book by id=" + id));
+        Book book = assertBookExistsById(id);
         return bookMapper.toDto(book);
     }
 
     @Override
-    public BookDto updateById(CreateBookRequestDto requestDto, Long id) {
-        Book book = bookRepository.findById(id)
-                .orElseThrow(() -> new EntityNotFoundException("Can't find book by id=" + id));
-        if (requestDto != null) {
-            String newIsbn = requestDto.isbn();
-            if (newIsbn != null && !newIsbn.equals(book.getIsbn())) {
-                validateIsbnUniqueness(newIsbn, id);
-                book.setIsbn(newIsbn);
-            }
-            book.setTitle(requestDto.title() != null && !requestDto.title().isEmpty()
-                    ? requestDto.title() : book.getTitle());
-            book.setAuthor(requestDto.author() != null && !requestDto.author().isEmpty()
-                    ? requestDto.author() : book.getAuthor());
-            book.setPrice(
-                    requestDto.price() != null
-                            ? requestDto.price().max(BigDecimal.ZERO) :
-                            book.getPrice()
-            );
-            book.setDescription(requestDto.description() != null
-                    ? requestDto.description() : book.getDescription());
-            book.setCoverImage(requestDto.coverImage() != null
-                    ? requestDto.coverImage() : book.getCoverImage());
+    public BookDto updateById(UpdateBookRequestDto requestDto, Long id) {
+        Book book = assertBookExistsById(id);
+        if (requestDto.isbn() != null && !requestDto.isbn().isEmpty()) {
+            validateIsbnUniqueness(requestDto.isbn(), id);
+            book.setIsbn(requestDto.isbn());
+        } else {
+            book.setIsbn(book.getIsbn());
         }
-        return bookMapper.toDto(bookRepository.save(book));
+        book.setTitle(requestDto.title() != null && !requestDto.title().isEmpty()
+                ? requestDto.title() : book.getTitle());
+        book.setTitle(requestDto.title() != null && !requestDto.title().isEmpty()
+                ? requestDto.title() : book.getTitle());

```

Comment:
You do the same twice
```suggestion
        book.setTitle(Objects.nonNull(requestDto.title()) && !requestDto.title().isEmpty()
                ? requestDto.title() : book.getTitle());
```